src/utils/geometry/removeClosePoints.ts

```diff
@@ -0,0 +1,82 @@
+import { Image } from '../..';
+
+import { Point } from './points';
+
+export interface RemoveClosePointsOptions {
+  /**
+   * The minimum distance between points in the returned filtered points. If the distance is less or equal to 0, no point is removed.
+   * @default `0`
+   */
+  distance?: number;
+  /**
+   * Shows what kind of extremum is being computed.
+   * @default `'maximum'`
+   */
+  kind?: 'minimum' | 'maximum';
+  /**
+   * Channel number of an image where the extremum should be found.
+   * @default `0`
+   */
+  channel?: number;
+}
+/**
+ * Finds extreme values of an image which are not stacked together.
+ * @param points - Array of points that should be combined to improve.
+ * @param image - Image which extrema are calculated from.
+ * @param options - FilterPointsOptions
+ * @returns Array of Points.
+ */
+export function removeClosePoints(
+  points: Point[],
+  image: Image,
+  options?: RemoveClosePointsOptions,
```

Comment:
Let's make this mandatory, and also make the `distance` and `kind` options mandatory.

It doesn't make sense to have defaults that do nothing.

Reply:
Would it make sense to make the `kind` option mandatory for `getExtrema` as well?

Reply:
I don't have a strong opinion. It certainly makes more sense to have a default for this option in `getExtrema` since it's kind of implied that this option exists.

In `removeClosePoints` it's really not obvious that such an options should exist

Reply:
Ok, then I believe the changes are done.